bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/connectivity/rawdata/LocalConnectRawDataBean.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal.connectivity.rawdata;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.binding.solax.internal.model.InverterType;
+import org.openhab.binding.solax.internal.util.GsonSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link LocalConnectRawDataBean} collects the raw data and the specific implementation to return the parsed data.
+ * If there are differences between the inverters probably would be wise to split the parsing in seprate class(es)
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class LocalConnectRawDataBean implements RawDataBean, InverterData {
+
+    private final Logger logger = LoggerFactory.getLogger(LocalConnectRawDataBean.class);
+
+    private @Nullable String sn;
+    private @Nullable String ver;
+    private int type;
+    private short @Nullable [] Data;
```

Comment:
You can fix the checkstyle warning **MemberNameCheck** like this:
```suggestion
    @SerializedName("Data")
    private short @Nullable [] data;
```

same in the line below.